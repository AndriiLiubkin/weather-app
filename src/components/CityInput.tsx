import React, { FormEvent } from "react";

interface CityInputProps {
  onGetWeather: () => void;
  onCityChange: (e: string) => void;
}

const CityInput = ({ onCityChange, onGetWeather }: CityInputProps) => {
  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    onGetWeather();
  };

  return (
    <div className="choose-country">
      <form onSubmit={(e) => handleSubmit(e)}>
        <input
          onChange={(e) => onCityChange(e.target.value)}
          placeholder="Enter City"
        />
        <button type="submit">Submit</button>
      </form>
    </div>
  );
};

export default CityInput;
