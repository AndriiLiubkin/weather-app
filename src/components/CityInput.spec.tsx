import React from "react";
import CityInput from "./CityInput";
import Enzyme, { shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

const onGetWeather = () => console.log("getWeather");

const onCityChange = () => console.log("City Change");

describe("<CityInput />", () => {
  it("should test CityInput component", () => {
    const wrapper = shallow(
      <CityInput onCityChange={onCityChange} onGetWeather={onGetWeather} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should handle onSubmit handler", () => {
    const onSubmitSpy = jest.fn();
    const wrapper = shallow(
      <CityInput onGetWeather={onSubmitSpy} onCityChange={onCityChange} />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.find("form").simulate("submit", {
      preventDefault: () => jest.fn(),
    });
    expect(onSubmitSpy).toHaveBeenCalled();
  });
});
