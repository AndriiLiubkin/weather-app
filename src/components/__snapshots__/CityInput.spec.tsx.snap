// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityInput /> should handle onSubmit handler 1`] = `ShallowWrapper {}`;

exports[`<CityInput /> should test CityInput component 1`] = `ShallowWrapper {}`;
